src/main/java/dev/sigstore/fulcio/client/FulcioValidator.java

```diff
@@ -0,0 +1,126 @@
+    }
+    EncodedKeySpec publicKeySpec = new X509EncodedKeySpec(section.getBase64DecodedBytes());
+    PublicKey ctfePublicKeyObj = keyFactory.generatePublic(publicKeySpec);
+
+    CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+    X509Certificate fulcioRootObj =
+        (X509Certificate)
+            certificateFactory.generateCertificate(new ByteArrayInputStream(fulcioRoot));
+
+    TrustAnchor fulcioRootTrustAnchor = new TrustAnchor(fulcioRootObj, null);
+    // this should throw the InvalidAlgorithmException a bit earlier
+    new PKIXParameters(Collections.singleton(fulcioRootTrustAnchor));
+
+    // TODO: this ctloginfo probably needs to be appropriately filled out
+    CTLogInfo ctLogInfo =
+        new CTLogInfo(ctfePublicKeyObj, "fulcio ct log", "intentionally-nonsense");
+    return new FulcioValidator(ctLogInfo, fulcioRootTrustAnchor);
+  }
+
+  private FulcioValidator(CTLogInfo ctLogInfo, TrustAnchor fulcioRoot) {
+    this.ctLogInfo = ctLogInfo;
+    this.fulcioRoot = fulcioRoot;
+  }
+
+  public void validateSct(SigningCertificate sc)
+      throws CertificateEncodingException, FulcioValidationException {
+    if (!(sc.getLeafCertificate() instanceof X509Certificate)) {
+      throw new RuntimeException(
+          "Encountered non X509 Certificate when validating SCT. Leaf certificate is "
+              + sc.getLeafCertificate().getClass());
+    }
+    CertificateEntry ce =
+        CertificateEntry.createForX509Certificate((X509Certificate) sc.getLeafCertificate());
+
+    // a result is returned here, but I don't know what to do with it yet
+    VerifiedSCT.Status status = ctLogInfo.verifySingleSCT(sc.getSct(), ce);
+    if (status != VerifiedSCT.Status.VALID) {
+      throw new FulcioValidationException("SCT could not be verified because " + status.toString());
+    }
+  }
+
+  public void validateCertChain(SigningCertificate sc) throws FulcioValidationException {
+    CertPathValidator cpv;
+    try {
+      cpv = CertPathValidator.getInstance("PKIX");
+    } catch (NoSuchAlgorithmException e) {
+      //
+      throw new RuntimeException(
+          "No PKIX CertPathValidator, we probably shouldn't be here, but this seems to be a system library error not a program control flow issue",
+          e);
+    }
+
+    PKIXParameters pkixParams;
+    try {
+      pkixParams = new PKIXParameters(Collections.singleton(fulcioRoot));
+    } catch (InvalidAlgorithmParameterException e) {
+      throw new RuntimeException(
+          "Can't create PKIX parameters for fulcioRoot. This should have been checked when generating a validator instance",
+          e);
+    }
+    pkixParams.setRevocationEnabled(false);
+
+    // these certs are only valid for 15 minutes, so find a time in the validity period
+    Date dateInValidityPeriod =
+        new Date(((X509Certificate) sc.getLeafCertificate()).getNotBefore().getTime() + 1000);
```

Comment:
Why adding time (+1000)?

Reply:
it was arbitrary amount of time. Not sure I have an actual reason, lemme test without.

Reply:
kk 1000 is unnecessary